src/RestClient.cs

```diff
@@ -45,28 +45,46 @@ private HttpWebRequest createRequest(string relativeUrl, HttpMethod method, stri
                 case "AUTH":
                     baseUrl = RazorpayClient.DefaultAuthUrl;
                     break;
+                case "POS":
+                    baseUrl = RazorpayClient.PosUrl;
+                    break;
                 default:
                     baseUrl = RazorpayClient.BaseUrl;
                     break;
             }
 
-            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(baseUrl + relativeUrl);
+            // Ensure proper URL construction with path separator
+            string fullUrl = baseUrl.TrimEnd('/') + "/" + relativeUrl.TrimStart('/');
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(fullUrl);
             request.Method = method.ToString();
             request.ContentLength = 0;
             request.ContentType = "application/json";
 
             string userAgent = string.Format("{0} {1}", RazorpayClient.Version, getAppDetailsUa());
             request.UserAgent = "razorpay-dot-net/" + userAgent;
 
-            if (RazorpayClient.Key != null && RazorpayClient.Secret != null)
+            if (host == "POS")

```

Comment:
instead of host pass authType with default value to Private Auth
AuthType = Public/Private